pkg/client/httputils.go

```diff
@@ -218,6 +218,9 @@ func (u *FilesUploader) Upload(files []string) error {
 		return err
 	}
 	ctx, cancel := context.WithCancel(context.Background())
+	// ensure cancel is called even in the case of an error
```

Comment:
Actually, the existing code only calls `cancel` when there are errors. This ensures it's called when there are NO errors too.